### ISSUE TITLE
Make the Problem List Ordered & Leverage GraphQL API

### ISSUE THREAD

**Issue by AerysNan** [LeetCode-OpenSource/vscode-leetcode#626]
I don't know whether it's a bug or a feature. So I post it as a feature proposal anyway.

## 🚀 Feature Proposal

Make the problems on the left sidebar ordered by their problem ID.

## Motivation

It may make people feel much better if they are ordered.

## Contribution

There are two ways to solve this problem:

1. We can sort the problem list after retrieving it from `LeetCodeExecutor.listProblems` by each problem's `fid`.
2. Or we can make the result coming from `leetcode-cli` ordered. I investigate the source code of `leetcode-tools/leetcode-cli` a little bit and suppose the function [`plugin.getProblems`](https://github.com/leetcode-tools/leetcode-cli/blob/cd5d4ef16105533fe29eef840fde14fd4958984f/lib/plugins/leetcode.js#L57) is the one that should be updated. After gathering all the problems from different categories. We can sort them by their `fid` attributed.

## Another issue

Besides, I notice that the `graphql` method in `leetcode-cli` is defined but not leveraged. Can we use `graphql` to get problems  a specific company or tag? So that we can abandon the `company.js` file to avoid hardcoding tag and company information (enven now this information is not updated frequently though).

---

The issue section of `leetcode-tools/leetcode-cli` is closed but I find that that repository and this one are maintained by the same group of people. So I post this here. I would appreciate it if you could kindly consider my proposal.

**Comment by yihong0618** [comment]
Yes, LeetCode is changing to `graphql `  setup by setup see #617. I find it would be wonderful if we change to `graphql ` too.

**Comment by AerysNan** [comment]
> @AerysNan
> Yes, LeetCode is changing to `graphql ` step by step see #617. I think it would be wonderful if we change to `graphql ` too.

@yihong0618 I see. So you're currently working on it? Or else I can write a PR for this (tag and company in GraphQL for a start).

**Comment by yihong0618** [comment]
> > @AerysNan
> > Yes, LeetCode is changing to `graphql ` step by step see #617. I think it would be wonderful if we change to `graphql ` too.
> 
> @yihong0618 I see. So you're currently working on it? Or else I can write a PR for this (tag and company in GraphQL for a start).

I am not. I think you can write a PR for that, I change [star problem](https://github.com/leetcode-tools/leetcode-cli/pull/49) to `graphql` recently

**Comment by AerysNan** [comment]
@yihong0618 Okay, I think I'll deal with it in a week or so. Then I'll create a PR for `leetcode-cli`.
